### PR TITLE
feat: swap out curve25519-dalek-ng for curve25519-dalek

### PR DESCRIPTION
**Pull request by hansieodendaal**
Swapped out `curve25519-dalek-ng` for `curve25519-dalek` in preparation for adding `tari-bulletproofs-plus` to `tari-crypto`

### CODE REVIEW

File: src/protocols.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        digest::Digest::update(&mut hasher, input);
+        Self::from_hash_sha3_512(hasher)
+    }
+
+    fn from_hash_sha3_512(hasher: Sha3_512) -> RistrettoPoint {
+        let output = hasher.finalize();
+        let mut output_bytes = [0u8; 64];
+        output_bytes.copy_from_slice(output.as_slice());
+
+        RistrettoPoint::from_uniform_bytes(&output_bytes)
```
Comment (sdbondi) [inline src/protocols.rs]:
```suggestion
        let output = hasher.finalize();
        RistrettoPoint::from_uniform_bytes(&output.into())
```

File: Cargo.toml [hunk 2]
```diff
@@ -17,8 +17,8 @@ description = "A pure-Rust implementation of Bulletproofs using Ristretto"
 edition = "2018"
 
 [dependencies]
-curve25519-dalek = { package = "curve25519-dalek-ng", version = "4.1", default-features = false, features = ["u64_backend", "serde", "alloc"] }
-subtle = { package = "subtle-ng", version = "2.5", default-features = false }
+curve25519-dalek = { package = "curve25519-dalek", version = "4.0.0-pre.2", default-features = false, features = ["serde", "alloc"] }
```
Comment (sdbondi) [inline Cargo.toml]:
```suggestion
curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["serde", "alloc"] }
```

**Review by sdbondi — COMMENTED** [review]
LGTM - some minor nits

**Review by sdbondi — APPROVED** [review]
Meant to approve in the last review because there are only nits

**Review by stringhandler — APPROVED** [review]

**Pull request merged**